hepcrawl/parsers/elsevier.py

```diff
@@ -0,0 +1,621 @@
+        self.builder.add_copyright(**self.copyright)
+        self.builder.add_document_type(self.document_type)
+        self.builder.add_license(**self.license)
+        for author in self.authors:
+            self.builder.add_author(author)
+        self.builder.add_number_of_pages(self.number_of_pages)
+        self.builder.add_publication_info(**self.publication_info)
+        for collab in self.collaborations:
+            self.builder.add_collaboration(collab)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        for keyword in self.keywords:
+            self.builder.add_keyword(keyword)
+        self.builder.add_imprint_date(self.publication_date.dumps())
+        for reference in self.references:
+            self.builder.add_reference(reference)
+
+        return self.builder.record
+
+    @property
+    def references(self):
+        """Extract a Elsevier record into an Inspire HEP references record.
+
+        Returns:
+            List[dict]: an array of reference schema records, representing
+                the references in the record
+        """
+        ref_nodes = self.root.xpath('.//bib-reference')
+        return list(
+            itertools.chain.from_iterable(
+                self.get_reference_iter(node) for node in ref_nodes
+            )
+        )
+
+
+    remove_tags_config_abstract = {
+        'allowed_tags': ['sup', 'sub'],
+        'allowed_trees': ['math'],
+        'strip': 'self::pub-id|self::issn'
+    }
+
+    @property
+    def abstract(self):
+        abstract_nodes = self.root.xpath(".//head/abstract/abstract-sec/simple-para")
+
+        if not abstract_nodes:
+            return
+
+        abstract = remove_tags(abstract_nodes[0], **self.remove_tags_config_abstract).strip('/ /n')
+        return abstract
+
+    @property
+    def article_type(self):
+        """Return a doctype mapped from abbreviation."""
+        abbrv_doctype = self.root.xpath(".//@docsubtype").extract()
+        doctype = ''
+        if abbrv_doctype:
+            doctype = DOCTYPE_MAPPING[get_first(abbrv_doctype)]
+        elif self.top_level_document_type:
+            doctype = self.top_level_document_type
+        if self.root.xpath("./conference-info"):
+            doctype = "conference paper"
+        if doctype:
+            return doctype
+
+    @property
+    def top_level_document_type(self):
+        article_path = "./*[self::article or self::simple-article]"
+        book_path = "./*[self::book or self:book-chapter or self::book-review or self::simple-book]"
+
+        if self.root.xpath(article_path):
+            doctype = "article"
+        elif self.root.xpath(book_path):
+            doctype = "book-review"
+        elif self.root.xpath("./exam"):
+            doctype = "exam"
+        return doctype
+
+    @property
+    def artid(self):
+        artid = self.root.xpath(
+            './{}/item-info/aid/text()'.format(self.top_level_document_type)
```

Comment:
not sure I understand this: if there's a `simple-article` node, it becomes `article`?